jaxrs-api/src/main/java/javax/ws/rs/JAXRS.java

```diff
@@ -0,0 +1,375 @@
+     * The configuration essentially consists of a set of parameters. While the set
+     * of actually effective keys is product specific, the key constants defined by
+     * the {@link JAXRS.Configuration} interface MUST be effective on all compliant
+     * products. Any unknown key MUST be silently ignored.
+     * </p>
+     *
+     * @author Markus KARG (markus@headcrashing.eu)
+     *
+     * @since 2.2
+     */
+    public static interface Configuration {
+
+        /**
+         * Configuration key for the protocol an application is bound to. A compliant
+         * implementation at least MUST accept the strings {@code "HTTP"} and
+         * {@code "HTTPS"} if these protocols are supported. The default value is
+         * {@code "HTTP"}.
+         */
+        static final String PROTOCOL = "javax.ws.rs.JAXRS.Protocol";
+
+        /**
+         * Configuration key for the hostname or IP address an application is bound to.
+         * If a hostname is provided, the application MUST be bound to <em>all</em> IP
+         * addresses assigned to that hostname. A compliant implementation at least MUST
+         * accept strings bearing hostnames, IP4 address strings, and IP6 address
+         * strings. The default value is {@code "localhost"}.
+         */
+        static final String HOST = "javax.ws.rs.JAXRS.Host";
+
+        /**
+         * Configuration key for the TCP port an application is bound to. A compliant
+         * implementation MUST accept {@code java.lang.Integer} values. There is no
+         * default value mandated by this specification. A compliant implementation MUST
+         * use its own defaults and MAY apply (but is not obligated to) auto-selection
+         * and range-scanning algorithms.
+         */
+        static final String PORT = "javax.ws.rs.JAXRS.Port";
+
+        /**
+         * Configuration key for the root path an application is bound to. The default
+         * value is {@code "/"}.
+         */
+        static final String ROOT_PATH = "javax.ws.rs.JAXRS.RootPath";
+
+        /**
+         * Returns the value of the property with the given name, or {@code null} if
+         * there is no property of that name.
+         *
+         * @param name
+         *            a {@code String} specifying the name of the property.
+         * @return an {@code Object} containing the value of the property, or
+         *         {@code null} if no property exists matching the given name.
+         */
+        Object property(String name);
+
+        /**
+         * Convenience method to get the {@code protocol} to be used.
+         * <p>
+         * Same as if calling {@link #property(String) property(PROTOCOL)}.
+         * </p>
+         *
+         * @return protocol to be used (e. g. {@code "HTTP")}.
+         * @throws ClassCastException
+         *             if protocol is not a {@link String}.
+         * @see JAXRS.Configuration#PROTOCOL
+         */
+        default String protocol() {
```

Comment:
I'm not sure if this was already discussed, but do we want to use `procotol()` or `getProtocol()`?

Reply:
*Getter* and *Setter* (in contrast to *Accessor* and *Mutator* - unfortunately much too often treated as synonyms in literature, while it is actually *not* a synonym) is a paradigm of Java *components* API aka "Beans" (*component property* method, vs. *class property* method), which is commonly understood as "returning *the same* that was given by `set`". For our use case this paradigm does not fit at all, as our configuration API allows to e. g. put "-1" and pull "8443" in turn, which confuses people used to "Beans" (in contrast to "Pojos"). Hence, I stick with the pure OOP naming variant, i. e. without `get`.